src/sourceTracking.ts

```diff
@@ -99,21 +131,22 @@ export class SourceTracking extends AsyncCreatable {
       .map((group) => group.filter((item) => this.forceIgnore.accepts(item)));
 
     // it'll be easier to filter filenames and work with smaller component sets than to filter SourceComponents
-    const groupings = (
-      byPackageDir
-        ? this.packagesDirs.map((pkgDir) => ({
-            path: pkgDir.name,
-            nonDeletes: allNonDeletes.filter((f) => pathIsInFolder(f, pkgDir.name)),
-            deletes: allDeletes.filter((f) => pathIsInFolder(f, pkgDir.name)),
-          }))
-        : [
-            {
-              nonDeletes: allNonDeletes,
-              deletes: allDeletes,
-              path: this.packagesDirs.map((dir) => dir.name).join(';'),
-            },
-          ]
-    ).filter((group) => group.deletes.length || group.nonDeletes.length);
+    const groupings = // if the users specified true or false for the param, that overrides the project config
```

Comment:
I find ternaries like this hard to grasp. Is there any way to encapsulate it so that it's easier to understand at first glance?

e.g.

```typescript
function isByPackageDir(byPackageDir, projectConfig) {
  return byPackageDir ?? getBoolean(projectConfig, 'pushPackageDirectoriesSequentially', false)
}

function getSequential() {
  return this.packagesDirs.map((pkgDir) => ({
    path: pkgDir.name,
    nonDeletes: allNonDeletes.filter((f) => pathIsInFolder(f, pkgDir.name)),
    deletes: allDeletes.filter((f) => pathIsInFolder(f, pkgDir.name)),
  }))
}

function getNonSequential() {
  return [
    {
      nonDeletes: allNonDeletes,
      deletes: allDeletes,
      path: this.packagesDirs.map((dir) => dir.name).join(';'),
    },
  ]
}

const groupings = isByPackageDir(byPackageDir, projectConfig) ? getSequential() : getNonSequential();
```

Or something like that